### PR TITLE
MQTT outbound connector doesn't use connection properties

### PR DESCRIPTION
**Pull request by vplechnoy**
See [#3814](https://devtopia.esri.com/WebGIS/arcgis-geoevent/issues/3814) for more details.

[x] Address an issue reported in verification https://devtopia.esri.com/WebGIS/arcgis-geoevent/issues/3814#issuecomment-3812792
[x] Improve MQTT transport reconnect logic
[x] Connector release version is not changed as requested https://devtopia.esri.com/WebGIS/arcgis-geoevent/issues/3814#issuecomment-3813826

GeoEvent build is available for verification at https://ps0016064.esri.com:6143/geoevent/manager

Screenshot 2022-11-22 at 9 58 36 AM

MQTT connection error

MQTT Release version 2

### CODE REVIEW

File: mqtt-transport/src/main/java/com/esri/geoevent/transport/mqtt/MqttTransportBase.java [hunk 1]
```diff
@@ -23,7 +23,7 @@
 
 public abstract class MqttTransportBase extends TransportBase implements MqttCallback, Runnable
 {
-  protected static final BundleLogger LOGGER = BundleLoggerFactory.getLogger(MqttTransportBase.class.getName(), "com.esri.ges.framework.transport.transport-api");
+  protected static final BundleLogger LOGGER = BundleLoggerFactory.getLogger(MqttTransportBase.class.getName(), "com.esri.geoevent.transport.mqtt-transport");
```
Comment (vplechnoy) [inline mqtt-transport/src/main/java/com/esri/geoevent/transport/mqtt/MqttTransportBase.java:26]: Invalid bundle symbolic name was causing a logger not to translate error messages correctly.

File: mqtt-transport/src/main/java/com/esri/geoevent/transport/mqtt/MqttTransportBase.java [hunk 2]
```diff
@@ -42,6 +42,8 @@ public String getName()
   public void afterPropertiesSet()
   {
     super.afterPropertiesSet();
+    // disconnect the transport before applying configuration changes
+    disconnect();
```
Comment (vplechnoy) [inline mqtt-transport/src/main/java/com/esri/geoevent/transport/mqtt/MqttTransportBase.java:46]: We should disconnect a transport here so the newly updated configuration is applied correctly.

File: mqtt-transport/src/main/java/com/esri/geoevent/transport/mqtt/MqttTransportBase.java [hunk 3]
```diff
@@ -274,8 +276,6 @@ private void readConfig()
 
   private void startTransport()
   {
-    if (mqttClientManager == null)
```
Comment (vplechnoy) [inline mqtt-transport/src/main/java/com/esri/geoevent/transport/mqtt/MqttTransportBase.java:277]: This logic is moved to 'connect()' method (see below).

File: mqtt-transport/src/main/java/com/esri/geoevent/transport/mqtt/MqttTransportBase.java [hunk 4]
```diff
@@ -302,7 +304,10 @@ private void connect() throws Exception
   private void disconnect()
   {
     if (mqttClientManager != null)
+    {
```
Comment (vplechnoy) [inline mqtt-transport/src/main/java/com/esri/geoevent/transport/mqtt/MqttTransportBase.java:307]: Disconnect logic also releases the resources in MQTT client manager.

**Review by hanoch — APPROVED** [review]

**Pull request merged**